content/doc/book/security/csrf-protection.adoc

```diff
@@ -0,0 +1,70 @@
+---
+title: CSRF Protection
+layout: section
+---
+ifdef::backend-html5[]
+:toc:
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+:hide-uri-scheme:
+endif::[]
+
+https://en.wikipedia.org/wiki/Cross-site_request_forgery[Cross-Site Request Forgery] (CSRF or XSRF) is a type of security vulnerability in web applications.
+Without protection from CSRF, a Jenkins user or administrator visiting some other web site would allow the operator of that site to perform actions in Jenkins as the victim.
+
+## CSRF Protection in Jenkins
+
+CSRF protection uses a token (called _crumb_ in Jenkins) that is created by Jenkins and sent to the user.
+Any form submissions or similar action resulting in modifications, like triggering builds or changing configuration, requires that the crumb be provided.
+The crumb contains information identifying the user it was created for, so submissions with another user's token would be rejected.
+All of this happens in the background and has no visible impact except in rare circumstances, e.g., after a user's session expired and they logged in again.
+
+NOTE: The documentation on this page applies to Jenkins 2.222 or newer.
+
+
+## Configuring CSRF Protection
+
+In _Manage Jenkins » Configure Global Security » CSRF Protection_, administrators can configure CSRF Protection.
+
+image::managing/configure-global-security-prevent-csrf.png["CSRF Protection in Configure Global Security", role=center]
+
+The _Default Crumb Issuer_ encodes the following information in the https://en.wikipedia.org/wiki/Cryptographic_hash_function[hash] used as crumb:
+
+* The user name that the crumb was generated for
+* The web session ID that the crumb was generated in
+* The IP address of the user that the crumb was generated for
+* A https://en.wikipedia.org/wiki/Salt_(cryptography)[salt] unique to this Jenkins instance
+
+All of this information needs to match when a crumb is sent back to Jenkins for that submission to be considered valid.
+
+The only supported option _Enable proxy compatibility_ removes information about the user IP address from the token.
+This can be useful when Jenkins is running behind a reverse proxy and a user's IP address as seen from Jenkins would regular change.
```

Comment:
```suggestion
This can be useful when Jenkins is running behind a reverse proxy and a user's IP address as seen from Jenkins would regularly change.
```